WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/sections/insights/usecases/ViewsAndVisitorsUseCase.kt

```diff
@@ -0,0 +1,277 @@
+    override fun buildLoadingItem(): List<BlockListItem> =
+            listOf(
+                    ValueItem(
+                            value = "0",
+                            unit = string.stats_views,
+                            isFirst = true,
+                            contentDescription = resourceProvider.getString(string.stats_loading_card)
+                    )
+            )
+
+    override suspend fun loadCachedData(): VisitsAndViewsModel? {
+        statsWidgetUpdaters.updateViewsWidget(statsSiteProvider.siteModel.siteId)
+        val cachedData = visitsAndViewsStore.getVisits(
+                statsSiteProvider.siteModel,
+                statsGranularity,
+                LimitMode.All
+        )
+        if (cachedData != null) {
+            logIfIncorrectData(cachedData, statsGranularity, statsSiteProvider.siteModel, false)
+            selectedDateProvider.onDateLoadingSucceeded(statsGranularity)
+        }
+        return cachedData
+    }
+
+    override suspend fun fetchRemoteData(forced: Boolean): State<VisitsAndViewsModel> {
+        val response = visitsAndViewsStore.fetchVisits(
+                statsSiteProvider.siteModel,
+                statsGranularity,
+                LimitMode.Top(OVERVIEW_ITEMS_TO_LOAD),
+                forced
+        )
+        val model = response.model
+        val error = response.error
+
+        return when {
+            error != null -> {
+                selectedDateProvider.onDateLoadingFailed(statsGranularity)
+                State.Error(error.message ?: error.type.name)
+            }
+            model != null && model.dates.isNotEmpty() -> {
+                logIfIncorrectData(model, statsGranularity, statsSiteProvider.siteModel, true)
+                selectedDateProvider.onDateLoadingSucceeded(statsGranularity)
+                State.Data(model)
+            }
+            else -> {
+                selectedDateProvider.onDateLoadingSucceeded(statsGranularity)
+                State.Empty()
+            }
+        }
+    }
+
+    /**
+     * Track the incorrect data shown for some users
+     * see https://github.com/wordpress-mobile/WordPress-Android/issues/11412
+     */
+    @Suppress("MagicNumber")
+    private fun logIfIncorrectData(
+        model: VisitsAndViewsModel,
+        granularity: StatsGranularity,
+        site: SiteModel,
+        fetched: Boolean
+    ) {
+        model.dates.lastOrNull()?.let { lastDayData ->
+            val yesterday = localeManagerWrapper.getCurrentCalendar()
+            yesterday.add(Calendar.DAY_OF_YEAR, -1)
+            val lastDayDate = statsDateFormatter.parseStatsDate(granularity, lastDayData.period)
+            if (lastDayDate.before(yesterday.time)) {
+                val currentCalendar = localeManagerWrapper.getCurrentCalendar()
+                val lastItemAge = ceil((currentCalendar.timeInMillis - lastDayDate.time) / 86400000.0)
+                analyticsTracker.track(
+                        STATS_OVERVIEW_ERROR,
```

Comment:
Should we create a seperate ERROR for `ViewsAndVisitors`? Then it wouldn't mix with overview error.

Reply:
Yes, probably when doing graph